### ISSUE TITLE
[MAINT]: issue_label doesn't work for Organizations

### ISSUE THREAD

**Issue by ikegentz** [integrations/terraform-provider-github#1840]
### Describe the need

There is currently no way to create an issue_label for a repo owned by an organization, rather than the account on which the PAT is used for authentication in the provider.

Specifically, the `repository` attribute in the `github_issue_label` resource is hard-coded to this format: `https://api.github.com/repos/<PAT user>/`, meaning if you try to pass `<my organization>/<repo>`, you'll end up with `https://api.github.com/repos/<PAT user>/<my organization>/<repo>`. 

[The API](https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#list-labels-for-a-repository) already supports an `owner` argument, so it seems like that could be exposed in the resource so that this would work.

### Relevant log output

```shell
│ Error: POST https://api.github.com/repos/<PAT user>/<my org>/<repo>/labels: 404 Not Found []
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kfcampbell** [comment]
Great catch! Do you have the interest/inclination to open up a PR to fix this behavior?

**Comment by ikegentz** [comment]
> Great catch! Do you have the interest/inclination to open up a PR to fix this behavior? 

Yes, I'd be happy to open a PR for this. May need a little guidance for getting started, but I should be able to get to this in a couple of weeks.

**Comment by kfcampbell** [comment]
Awesome! Our [CONTRIBUTING.md](https://github.com/integrations/terraform-provider-github/blob/main/CONTRIBUTING.md), especially [this section](https://github.com/integrations/terraform-provider-github/blob/main/CONTRIBUTING.md#building-the-provider) about how to build the provider so it's invoked in your local Terraform runs. For full debugging, [this section](https://github.com/integrations/terraform-provider-github/blob/main/CONTRIBUTING.md#debugging-the-terraform-provider) is the one to read.

**Comment by ikegentz** [comment]
Hey @kfcampbell apologies for such a delay in taking a look at this. I've begun working on this but had a question w.r.t. running the initial build, is it normal to have to set `mod=mod` to get `make build` to work? I kept running into:

```
github/resource_github_issue_labels.go:8:2: cannot find module providing package github.com/google/go-github/v52/github: import lookup disabled by -mod=vendor
        (Go version in go.mod is at least 1.14 and vendor directory exists.)
```
when running `make build`. Not a big deal as I was able to get past this, but if this should be noted in the `Contributing` guide I'd be happy to make a note!
